clients/devtools-entry.js

```diff
@@ -67,6 +68,70 @@ function listenForStatus(listenCallback) {
   log.events.addListener('status', listenCallback);
 }
 
+/**
+ * With just a trace, provide Lighthouse performance report
+ * From DevTools it'll look something like this
+ *
+    self.analyzeTrace({traceEvents: tEvents}, {url: 'http://page', device: 'desktop'})
+      .then(console.log)
+      .catch(console.warn);
+ *
+ * @param {LH.Trace} trace
+ * @param {{device: "mobile" | "desktop", url: string}} opts
+ * @return {Promise<LH.RunnerResult|undefined>}
+ */
+function analyzeTrace(trace, opts) {
+  const url = opts.url;
+
+  /** @type {LH.Config.Json} */
+  const configJSON = {
+    extends: 'lighthouse:default',
+    settings: {
+      onlyCategories: ['performance'],
+      output: ['json'],
+      formFactor: opts.device,
+      throttlingMethod: 'devtools', // can't do lantern right now, so need real throttling applied.
+      // In DevTools, emulation is applied _before_ Lighthouse starts (to deal with viewport emulation bugs). go/xcnjf
+      // As a result, we don't double-apply viewport emulation.
+      screenEmulation: {disabled: true},
+      traceBasedNetworkRecords: true,
```

Comment:
as an alternative api....

What if the settings took the trace directly? `_internalProvidedTrace` then devtools would just call `self.runLighthouse` with a particular config (possibly by calling `self.createConfig` first then adding the special option).

Reply:
Uh... nevermind. I overlooked that this function isn't actually running lighthouse gathering.